content/posts/2019-08-03---Going-from-ember-to-vue.md

```diff
@@ -0,0 +1,284 @@
+<aside>Photo by <a href="https://www.pexels.com/@hilaryh">Hilary Halliwell on Pexels</a></aside>
+
+This isn't meant to be a knock on Vue, but it doesn't come with as much stuff out of the box as Ember.
+In fact, this is probably a huge plus.
+[Vue CLI](https://cli.vuejs.org/) allows you to generate a project quickly, but much like Create React App, you are getting a pretty simple UI application with webpack pre-configured for you.
+And yes, webpack over Brocolli is another big difference... but it's probably a good one.
+
+However, anything you want to do in ember, probably already has an equivalent library in Vue.
+And most of these libraries are maintained by core team members, so they feel very official.
+With just a couple more `vue add` commands, you can get routing and state management in your app, so it's really not a _huge_ knock against Vue.
+
+## Components are one file
+
+I found this weird at first, but boy, do I like components with both markup and JavaScript in the same file.
+I hated having to bounce between files so often in Ember (especially when the community wanted to move away from pod structure).
+
+<figure>
+  <blockquote>
+    <p>Ugh, markup and JS in the same file?? I believe in separation of concerns.</p>
+    <footer>
+      <cite>— People who want you to know they're smart.</cite>
+    </footer>
+  </blockquote>
+</figure>
+
+I've actually heard this complaint a few times about "I don't like JS/HTML/CSS in the same file because I like separation of concerns."
+I get the sentiment, but to be clear, just because "concerns" are in the same file, doesn't mean they aren't separated.
+The JavaScript of a component is too tightly tied to the markup (as it should be), so why force yourself to bounce between the two files?
+
+## State management
+
+The popular state management solution seems to be [Vuex](https://vuex.vuejs.org/) (another "official" Vue library written by core team members).
+This is much closer to [Redux](https://redux.js.org/) or [NGRX](https://ngrx.io/).
+I'm not going to claim that these patterns are the end-all-be-all solutions for state management on the front-end, but it's where most of the community seems to be going right now.
+Which is a huge plus over Ember.
+The skills/practices you pick up writing Vuex will translate to these other libraries.
+
+You are also much more explicit when updating your "store" and that makes it feel so much less magical than Ember Data.
+No, there's no identity mapping or convenient CRUD methods, but that also means you get to be more flexible with how you request data (and as a result, you'll probably be much more aware of what is going on in your application).
+
+# Similarities
+
+![Spider-Man pointing to Spider-Man](https://i.kym-cdn.com/entries/icons/mobile/000/023/397/C-658VsXoAo3ovC.jpg)
+
+Ok, let's talk about the similarities (which is really the point of this post).
+
+## Components
+
+Components in Vue are more similar to Ember than the ones in React.
```

Comment:
Is there a different under the hood?  Glimmer with binary format + eliminating parse step vs not sure what is on Vue's side.